### ISSUE TITLE
End user JWT authentication is not working as expected with a OAuth2 Server running within the k8s cluster

### ISSUE THREAD

**Issue by zhaorendong** [istio/proxy#2033]
I am try use End user JWT authentication  with a OAuth2 Server, I created policy as follows. But I tried to test visits, the http_code is always 200.
apiVersion: "authentication.istio.io/v1alpha1"
kind: "Policy"
metadata:
  name: "jwt-example"
spec:
  targets:
  - name: httpbin
  origins:
  - jwt:
      issuer: "testing@secure.istio.io"
      jwksUri: "http://localhost:1111/Ku8Master/test"
  principalBinding: USE_ORIGIN

[2018-11-09 06:54:44.082][25][info][config] ./src/envoy/http/jwt_auth/auth_store.h:72] Loaded JwtAuthConfig: rules {
  issuer: "https://securetoken.google.com"
  audiences: "productpage"
  local_jwks {
    inline_string: "{\"keys\":[{\"kty\":\"RSA\",\"e\":\"AQAB\",\"kid\":\"DHFbpoIUqrY8t2zpA2qXfCmr5VO5ZEr4RzHU_-envvQ\",\"n\":\"xAE7eB6qugXyCAG3yhh7pkDkT65pHymX-P7KfIupjf59vsdo91bSP9C8H07pSAGQO1MV_xFj9VswgsCg4R6otmg5PV2He95lZdHtOcU5DXIg_pbhLdKXbi66GlVeK6ABZOUW3WYtnNHD-91gVuoeJT_DwtGGcp4ignkgXfkiEm4sw-4sfb4qdt5oLbyVpmW6x9cfa7vs2WTfURiCrBoUqgBo_-4WTiULmmHSGZHOjzwa8WtrtOQGsAFjIbno85jp6MnGGGZPYZbDAa_b3y5u-YpW7ypZrvD8BgtKVjgtQgZhLAGezMt0ua3DRrWnKqTZ0BJ_EyxOGuHJrLsn00fnMQ\"}]}"
  }
  forward: true
  from_headers {
    name: "x-goog-iap-jwt-assertion"
  }
  forward_payload_header: "istio-sec-4cd586ece11fcc8dbb78ad8b1188c7369e6618d8"
}
allow_missing_or_failed: true

**Comment by qiwzhang** [comment]
Hi @zhaorendong 

Could you dump your envoy config?  by 

curl http://127.0.0.1:admin_port/config_dump

Also could you post your expired JWT token here too

**Comment by zhaorendong** [comment]
A part of envoy config as follows.There is another problem, if I post expired JWT token,does the envoy not be intercepted? In other words, I must post effective JWT token,  the http_code is 401?

         "http_filters": [
            {
             "name": "jwt-auth",
             "config": {
              "allow_missing_or_failed": true,
              "rules": [
               {
                "forward": true,
                "issuer": "testing@secure.istio.io",
                "local_jwks": {
                 "inline_string": "{\"keys\":[{\"kty\":\"RSA\",\"e\":\"AQAB\",\"kid\":\"DHFbpoIUqrY8t2zpA2qXfCmr5VO5ZEr4RzHU_-envvQ\",\"n\":\"xAE7eB6qugXyCAG3yhh7pkDkT65pHymX-P7KfIupjf59vsdo91bSP9C8H07pSAGQO1MV_xFj9VswgsCg4R6otmg5PV2He95lZdHtOcU5DXIg_pbhLdKXbi66GlVeK6ABZOUW3WYtnNHD-91gVuoeJT_DwtGGcp4ignkgXfkiEm4sw-4sfb4qdt5oLbyVpmW6x9cfa7vs2WTfURiCrBoUqgBo_-4WTiULmmHSGZHOjzwa8WtrtOQGsAFjIbno85jp6MnGGGZPYZbDAa_b3y5u-YpW7ypZrvD8BgtKVjgtQgZhLAGezMt0ua3DRrWnKqTZ0BJ_EyxOGuHJrLsn00fnMQ\"}]}"
                },
                "forward_payload_header": "istio-sec-5406b7840708063f65cbdf52153ca364a476d68b"
               }
              ]
             }
            },
            {
             "config": {
              "policy": {
               "principal_binding": "USE_ORIGIN",
               "origins": [
                {
                 "jwt": {
                  "jwks_uri": "http://10.255.242.225:1111/Ku8Master/test",
                  "issuer": "testing@secure.istio.io"
                 }
                }
               ]
              },

**Comment by qiwzhang** [comment]
What I mean was: you post your expired token here for debugging.  we can use jwt.io to inspect it.  Not to use it in your test.

@lei-tang @diemtvu .  It could not problem in authentication filter.  Could you take a look?

**Comment by lei-tang** [comment]
@zhaorendong Can you share the detailed steps to reproduce the issue you encountered?

**Comment by zhaorendong** [comment]
Hi @lei-tang 
First, I created one namespaces foo,with two services, httpbin and sleep, both running with an Envoy sidecar proxy. I send an HTTP request with curl from the sleep pod to the httpbin in the namespace foo .
All requests succeed with HTTP code 200. This is all right. Then  I try to create End-user authentication According to this "https://istio.io/docs/tasks/security/authn-policy/#end-user-authentication". The only difference is that the issuer: "testing@secure.istio.io" and   jwksUri: "https://raw.githubusercontent.com/istio/istio/release-1.0/security/tools/jwt/samples/jwks.json" I can't connected. I don't know if it will affect that request with curl from the sleep pod to the httpbin in the namespace foo  always requests 200. My expectation is that when I used end-user-authentication, even if the issuer and jwksUri are not right, it would request with HTTP code 401.
